tests/DataValues/QuantityValueTest.php

```diff
@@ -138,10 +139,14 @@ public function newFromNumberProvider() {
 		$value = new DecimalValue( '-0.05' );
 		$value1 = new DecimalValue( '-0.04' );
 		$value2 = new DecimalValue( '-0.06' );
+		$expectedValue = new DecimalValue( $value->getValueFloat() );
+		$expectedValue1 = new DecimalValue( $value1->getValueFloat() );
+		$expectedValue2 = new DecimalValue( $value2->getValueFloat() );
+
 		$unit = 'm';
 		yield [
 			$value->getValueFloat(), $unit, $value1->getValueFloat(), $value2->getValueFloat(),
-			new QuantityValue( $value, $unit, $value1, $value2 )
+			new QuantityValue( $expectedValue, $unit, $expectedValue1, $expectedValue2 )
 		];
```

Comment:
I'm not sure we should change the test content like that, what's happening now (afaict) is that we are passing the expected result into the constructor, by essentially assuming the result we expect to get. What issue were you trying to resolve by this?

Reply:
I think the variable name "$expectedValue is not a good one" actually the test had a small bug and for some reason was passing on phpunit version 4 undetected. the actual computation changes a decimal to a float then converts the float output to decimal again. but the expected computation change to decimal and return the decimal. Those 2 computations give different values when the input is a negative decimal. So I change the expected computation to be similar to the actual by adding a conversion to float before converting to Decimal. I believe the variable name is confusing here. Maybe this long comment is also confusing. :(

We can jump in a call and I show you what I mean if not clear.

Reply:
I still don’t understand what is happening here, or what the difference between variables like `$value` and `$decimalValue` is supposed to be when they’re both instances of `DecimalValue` :/

Reply:
I changed the variable name to `decimalValueFromFloat`. Is that better?